WebKitBrowser/WebKitImplementation.cpp

```diff
@@ -1841,15 +2240,11 @@ static GSourceFuncs _handlerIntervention =
 
```

Comment:
Keep the logging and killHelper, we need to kill a particular thread in the child process, DeactivateBrowser is not enough.

Reply:
That is accommodated now by the core. Logging and killing is triggered from the core so that *all* app engines start behaving the same. The BreakPad and the loglines are coming from there. See: https://github.com/rdkcentral/Thunder/commit/7ed0e2b51bdf03a2a3043d9b44961d3b5248af2e#diff-2fd8d9c4de023db5b7ed89e1fd8562b2

Reply:
will it send a kill signal to a UI thread in the web process?

Reply:
The order is: Send SIGFPE (2) (for breakpad core dump generation), SIGTERM (6) (to ask for a "clean shutdown), SIGKILL (9). These signals should be forwarded to child processes, Network and Web  (UI process will definitily get it). But good to check

Reply:
So you kill all the processes? how do you know which one was hanging then? The idea for the original change was to kill just the hanging process (**send a signal to a specific thread in case of web process**) and get a crash report with details.

Reply:
The kill of the out-of-process plugin Is the same as it was. At the end the process is getting the same signals. The question is what happens with the WebProcess, which is part of the WPEProcess that spans the UI. 
With respect to that I consulted the WebKit team (Igalia) and there feedback: "@Pierre, there's WKPageTerminate() call which also dispatches SIGKILL, but also internally (inside UIProcess) manages all the relevant state more cleanly than if SIGKILL is just dispatched in the WebProcess's general direction"
So by definition the sending of the SIGFPE or SIGKILL might cause havoc in the parent process (relevant state is not cleared). To make sure that the relvant state is set correctly I prefer to take down the whole WebKit (For sure all state will be consistent :-) ) 
We could upon receiving the SIGFPE in the WPEProcess, forward this to the WebProcess, if it is hanging and if indeed it does not generate a callstack.By definition, if we would like the WPEProcess to handle the restart of the WebProcess, we must use the WKPageTerminate() to kill (after the SIGFPE for Breakpad post mortom generation).
Thinking of this do you include Breakpad in the build of WPEWebKit, cause the WebProcess is *not* spawned by the WPEProcess which has breakpad additions, the WebProcess is spawned on its own...

Reply:
The point is not just to kill the web process. but kill it in a way that we can track the crash on crashportal (as we discussed this already). WKPageTerminate is no good here...

Reply:
According to the Webkit developers *not* using WKPageTerminate is risco. So if you are not goign for a full restart of the WebKit you *must* call WKPageTerminate to make sure that the state of the WEBUI process is correct otherwise you have an inconsitsent state of the WebKit.

Reply:
I wonder if BreakPad is actually effective on WPEWebProcess (webkit) as there is *no* breakpad stuff in webkit, only in WPEProcess (thunder). Assuming it would work, than Thunder can not forward this request, it has to be in this plugin. Happy to add that, it would mean installing a SIGFPE handler to forward the request to the WPEProcess (partial shutdown)

Reply:
> According to the Webkit developers not using WKPageTerminate is mandatory to get the WEBUI process in a consistant state. So if you are not going for a full restart of the WebKit you must call WKPageTerminate to make sure that the state of the WEBUI process is correct otherwise you have an inconsitsent state of the WebKit.

... This is not just about killing or terminating the process. this is about sending a signal to a specific thread in the web process (WEBUI). WebKit is capable of handling crashed processes... 

And I'm totally missing the point about enforced usage of 'WKPageTerminate' and the consistent state of WEBUI. The process will be dead in either case. If this is about sending SIGKILL, I agree it can be replaced with 'WKPageTerminate' if that is the problem you trying to address here. The important part is sending SIGFPE to a particular thread...

> I wonder if BreakPad is actually effective on WPEWebProcess (webkit) as there is no breakpad stuff in webkit, only in WPEProcess (thunder). 
there is, in wpe webkit. see https://github.com/WebPlatformForEmbedded/WPEWebKit/blob/wpe-2.22/Source/WebKit/WebProcess/EntryPoint/unix/WebProcessMain.cpp